### PR TITLE
Adding Simmate from PyPI

### PR DESCRIPTION
**Pull request by jacksund**
(first-time contributor here -- I'm still getting the hang of publishing packages)

This is the original commit and still needs to be reviewed.

There are comments in my meta.yaml where I'm uncertain, and if anyone is able to check my work, it would be greatly appreciated!

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jacksund** [comment]
The windows build is failing because of matminer's conda-forge -- as their feedstock is failing on windows as well. I believe it's because [they require dscribe >=0.2.9,](https://github.com/conda-forge/matminer-feedstock/blob/738809bfc8b06e329c5382fe46c080707e8d4053/recipe/meta.yaml#L41) even though [dscribe only has 0.2.8 available on windows](https://anaconda.org/conda-forge/dscribe). I've mentioned this to the matminer team as well - [that thread is here](https://matsci.org/t/conda-install-broken/3427/26?u=jacksund).

**Comment by jacksund** [comment]
@conda-forge-admin, please ping team

This pull request is ready for review!

File: recipes/simmate/meta.yaml
Comment (ngam) [inline recipes/simmate/meta.yaml:71]: Hm... I wonder if there is a policy about this... probably not.

Comment (jacksund) [reply recipes/simmate/meta.yaml:71]: You mean making my repo public before giving it a BSD license? I can do that if you need.

Comment (jacksund) [reply recipes/simmate/meta.yaml:71]: I just made it public if that helps 😄. 
https://github.com/jacksund/simmate

**Comment by jacksund** [comment]
@ngam thank you for reviewing this. Is there anything else you'd like me to fix before merging?

**Comment by ngam** [comment]
> @ngam thank you for reviewing this. Is there anything else you'd like me to fix before merging?

To be clear, I cannot merge (I am not part of official reviews). However, if this were an actual review, you'd need to either accept or reject the suggested edits. Do you not want to try to make this into a noarch package?

**Comment by jacksund** [comment]
> To be clear, I cannot merge (I am not part of official reviews). 

Ah, sorry about that. I'm still trying to figure a lot of this stuff out

> However, if this were an actual review, you'd need to either accept or reject the suggested edits.

Where do I see these? I only see my commit, and I'm not allowed to accept my own additions.

> Do you not want to try to make this into a noarch package?

I'll give this a go and then re-request review when it's ready!

**Review by ngam — CHANGES_REQUESTED** [review]

**Comment by ngam** [comment]
> Where do I see these? I only see my commit, and I'm not allowed to accept my own additions.
> 
> >

OOOPS Sorry... Obviously, I am also a novice. I forgot to submit my review!

File: recipes/simmate/meta.yaml
Comment (ngam) [reply recipes/simmate/meta.yaml:25]: This is resolved. Feel free to mark it as such.

**Comment by ngam** [comment]
Also, simply try deleting the second compiler and see how that goes. It may well work, especially that it seems you're not building the dependency, right?

**Comment by jacksund** [comment]
haha all good. Thanks for the quick reply and sending those. I'll go through them now.

> you're not building the dependency, right?
Yep that's correct. I'll give this a try.

File: recipes/simmate/meta.yaml
Comment (jacksund) [reply recipes/simmate/meta.yaml:32]: Yep, I agree this needs to be fixed. My test suite is sorely lacking in the repo right now, so I'm starting strict with the dependencies and loosening as implement my tests. 

For my team's alpha testing, I'm making things really strict as I know it will be the only thing in their conda environments.

**Comment by ngam** [comment]
Looks good! This way, you only care about the linux_64 builds. The others are meaningless for now. This should pass. I would call the team again to review. They may review it faster since i gave it a quick look. I found that @dopplershift is really helpful when it comes to python pkgs (maybe can give more advice) but hey @conda-forge/staged-recipes this seem ready!

**Review by ngam — APPROVED** [review]

File: recipes/simmate/meta.yaml
Comment (ngam) [reply recipes/simmate/meta.yaml:29]: Yep, that's okay. I didn't notice that this was a `.dev` release at first. So it should be fine.

Comment (jacksund) [reply recipes/simmate/meta.yaml:29]: Sweet, that's good to hear. I'll mark this as resolved then.

**Comment by jacksund** [comment]
@ngam Thank you so much! This means a lot :smile:. Enjoy the holidays!

**Review by beckermr — CHANGES_REQUESTED** [review]
Dev releases need to be made to a separate channel so this recipe needs a conda build config yaml file with the right contents. I don’t have the link handy but it should be in our docs.

**Comment by ngam** [comment]
@beckermr I am not sure if this is strictly needed as this package hasn't been released in any other way (stable release etc) yet, so this could just start as this and develop into a stable release. This is the link to the docs: https://conda-forge.org/docs/maintainer/knowledge_base.html#pre-release-builds

**Comment by ngam** [comment]
Or maybe the worry is that this may break the user experience because it is a dev release and thus it is misleading to have in the main releases?

**Comment by beckermr** [comment]
We are not concerned about quality of the package itself. This is a standard policy we enforce generally. If there are no non-dev releases, then we can merge to the main channel but we request the dev ones get moved to a dev label later.

**Comment by ngam** [comment]
> We are not concerned about quality of the package itself. This is a standard policy we enforce generally. If there are no non-dev releases, then we can merge to the main channel but we request the dev ones get moved to a dev label later.

Thanks for the prompt reply! Yep, that was my understanding, thanks!

@jacksund I guess it would be up to you --- if you do a dev release in conda-forge, you cannot simply `conda install -c conda-forge simmate` and instead you'd need to specify a label/channel to install your package. You just have to make sure you don't push dev and non-dev at the same time to the main branch of the feedstock. You did have the github private until we talked above, so maybe you would be interested in the extra label here...

**Comment by jacksund** [comment]
Sorry for the slower reply. I'm trying to read the docs and understand your suggestions.

It seems straightforward to add a label, so I'm willing to go that route. The one thing I'm not seeing in the docs is how to specify a label when installing. Can you doublecheck my understanding in how this would work? 

I add this to the recipe:
```
channel_targets:
  - conda-forge simmate_dev
```

Then my package is installed via:
```
conda install -c conda-forge simmate_dev
```

And when I'm making stable releases, I simple remove the `channel targets` from my recipe, right?

**Comment by ngam** [comment]
> And when I'm making stable releases, I simply remove the `channel targets` from my recipe, right?

You can maintain both a dev and stable release together; the stable one would go into main/master and the dev release will have its separate branch (dev) and they can then both progress separate from each other. However, you may then decide you just want to release the stable release only on conda-forge (main branch). It will be clearer once you have the feedstock set up :)

**Comment by jacksund** [comment]
@ngam gotcha, thank you. I'm looking at the matplotlib feedstock and now everything is making sense.

**Comment by ngam** [comment]
(As an example, see this feedstock https://github.com/conda-forge/pyferret-feedstock where we have a dev branch as well as master branch)

**Comment by jacksund** [comment]
@beckermr I added the label to the build config. Hopefully this looks good now 🤞

**Pull request merged**
